### PR TITLE
Implement missing features for campaignion_foundation

### PR DESCRIPTION
**Pull request by torotil**

### CODE REVIEW

File: theming.theme.inc [hunk 1]
```diff
@@ -133,9 +133,9 @@ function theming_form_element($variables) {
   if (!empty($element['#description'])) {
     $output .= '<div' . drupal_attributes($element['#description_attributes']) . '>' . $element['#description'] . "</div>\n";
   }
-
-  $output .= "</div>\n";
-
+  if (empty($element['#wrapper_enabled'])) {
```
Comment (katzmo) [inline theming.theme.inc]: Why if empty? Don’t we want a wrapper `if ($element['#wrapper_enabled'])`?

File: theming.module [hunk 2]
```diff
@@ -17,10 +17,13 @@ function theming_theme_registry_alter(&$hooks) {
...
   foreach ($override as $name) {
     $hook = &$hooks[$name];
-    $hook['theme path'] = $path;
-    $hook['file'] = $file;
+    // Only replace the theme function when no theme overrides it.
+    if ($hook['function'] == "theme_$name") {
+      $hook['theme path'] = $path;
+      $hook['file'] = $file;
+      $hook['function'] = "theming_$name";
+    }
```
Comment (katzmo) [inline theming.module:25]: Does that mean when there is `my_theme_form_element` it overrides `theming_form_element`? (And before the fix `my_theme_form_element` would have been irgnored?)

File: theming.theme.inc (hunk 1 above)
Comment (torotil) [reply theming.theme.inc]: Ah yes. Seems I forgot to rewrite this condition after I had reintroduced the flag. Fixed in a new commit.

File: theming.module (hunk 2 above)
Comment (torotil) [reply theming.module:25]: Exactly. That’s why the `<span>` wasn’t applied.

Comment (katzmo) [reply theming.module:25]: Thanks!

**Review by katzmo — APPROVED** [review]

**Review by katzmo — APPROVED** [review]

**Pull request merged**